api/backend_test.go

```diff
@@ -90,50 +90,41 @@ func setupTestMultiDB() (*multiaccounts.Database, func() error, error) {
 	}, nil
 }
 
-func setupGethStatusBackend() (*GethStatusBackend, func() error, func() error, func() error, error) {
+func setupGethStatusBackend() (*GethStatusBackend, func(t *testing.T) error, error) {
 	db, stop1, err := setupTestDB()
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, nil, err
 	}
 	backend := NewGethStatusBackend()
 	backend.StatusNode().SetAppDB(db)
 
 	ma, stop2, err := setupTestMultiDB()
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, nil, err
 	}
 	backend.StatusNode().SetMultiaccountsDB(ma)
 
 	walletDb, stop3, err := setupTestWalletDB()
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, nil, err
 	}
 	backend.StatusNode().SetWalletDB(walletDb)
 
-	return backend, stop1, stop2, stop3, err
+	return backend, func(t *testing.T) error {
+		require.NoError(t, stop1())
+		require.NoError(t, stop2())
+		require.NoError(t, stop3())
+		require.NoError(t, backend.StopNode())
```

Comment:
You can use `t.Cleanup` for this:
```go
t.Cleanup(func() {
    stop1()
    stop2()
    stop3()
    backend.StopNode()
})
```

And no need to return custom callback then 🙌

Reply:
Updated.